cli/cmd/top.go

```diff
@@ -0,0 +1,393 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/linkerd/linkerd2/controller/api/util"
+	pb "github.com/linkerd/linkerd2/controller/gen/public"
+	"github.com/linkerd/linkerd2/pkg/addr"
+	runewidth "github.com/mattn/go-runewidth"
+	termbox "github.com/nsf/termbox-go"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+type topOptions struct {
+	namespace   string
+	toResource  string
+	toNamespace string
+	maxRps      float32
+	scheme      string
+	method      string
+	authority   string
+	path        string
+}
+
+type topRequest struct {
+	event   *pb.TapEvent
+	reqInit *pb.TapEvent_Http_RequestInit
+	rspInit *pb.TapEvent_Http_ResponseInit
+	rspEnd  *pb.TapEvent_Http_ResponseEnd
+}
+
+type topRequestID struct {
+	src    string
+	dst    string
+	stream uint64
+}
+
+type tableRow struct {
+	by          string
+	source      string
+	destination string
+	count       int
+	best        time.Duration
+	worst       time.Duration
+	last        time.Duration
+	successes   int
+	failures    int
+}
+
+const headerHeight = 3
+
+var (
+	columnNames  = []string{"Source", "Destination", "Path", "Count", "Best", "Worst", "Last", "Success Rate"}
+	columnWidths = []int{23, 23, 55, 6, 6, 6, 6, 3}
```

Comment:
I don't think it's worth fixing as part of this branch, but just noting that the destination column can sometimes be a pod name instead of an IP, and most pod names won't fit in 23 characters. For instance:

![image](https://user-images.githubusercontent.com/9226/44164924-3b88a580-a07c-11e8-9514-4bdfdd8c07bd.png)

This relates to https://github.com/linkerd/linkerd2/issues/1170#issuecomment-413267112 and https://github.com/linkerd/linkerd2/pull/1454#discussion_r210152291